### PR TITLE
[1LP][RFR] Services catalog collections

### PR DESCRIPTION
**Pull request by rsnyman**
Convert cfme/services/catalogs/catalog.py to collections

### CODE REVIEW

File: cfme/tests/services/test_different_dialogs_in_catalogs.py [hunk 1]
```diff
@@ -52,11 +51,9 @@ def tagcontrol_dialog(appliance):
 
 
 @pytest.yield_fixture(scope="function")
-def catalog():
+def catalog(appliance):
     catalog = "cat_" + fauxfactory.gen_alphanumeric()
-    cat = Catalog(name=catalog,
-                  description="my catalog")
-    cat.create()
+    cat = appliance.collections.catalogs.create(name=catalog, description="my catalog")
```
Comment (sshveta) [inline cfme/tests/services/test_different_dialogs_in_catalogs.py]: This change needs to be made in these places too : 
https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/services/test_catalog.py#L19
https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/services/test_provision_stack.py#L52
https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/services/test_dynamicdd_dialogelement.py#L74
https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/ansible/test_embedded_ansible_services.py#L110

Comment (rsnyman) [reply cfme/tests/services/test_different_dialogs_in_catalogs.py]: Thanks! Not sure how I missed those.

**Comment by rsnyman** [comment]
@sshveta Can you re-review while we wait for the tests to finish running please?

**Comment by sshveta** [comment]
LGTM :+1: 
Waiting for PRT results .

**Comment by rsnyman** [comment]
I'm seeing the following error in PRT:

```
>           raise TimedOutError("Could not do {} at {}:{} in time".format(message, filename, line_no))
E           TimedOutError: Could not do appliance.is_embedded_ansible_running at /cfme_tests_te/cfme/utils/appliance/__init__.py:1749 in time

/usr/lib/python2.7/site-packages/wait_for/__init__.py:180: TimedOutError
TimedOutError
Could not do appliance.is_embedded_ansible_running at /cfme_tests_te/cfme/utils/appliance/__init__.py:1749 in time
```

@mshriver @sshveta any idea why I'm getting this? I've checked the code, and I haven't touched that.

**Comment by sshveta** [comment]
Failures don't seem related to changes .
 @lkhomenk might be aware of test_provision_stack failures 
and @quarckster about 
Could not do appliance.is_embedded_ansible_running at /cfme_tests_te/cfme/utils/appliance/__init__.py:1749 in time

**Comment by mshriver** [comment]
WIP for merge conflict

**Comment by rsnyman** [comment]
I've checked the tests, and the failures appear to be happening in master too.

**Comment by rsnyman** [comment]
@mshriver ping

File: cfme/services/catalogs/catalog.py [hunk 2]
```diff
@@ -137,41 +126,65 @@ def exists(self):
...
+        view = navigate_to(self, 'Add')
+        view.fill({
+            'name': name,
+            'description': description,
+            'assign_catalog_items': items
+        })
+        view.add_button.click()
+        catalog = self.instantiate(name=name, description=description, items=items)
+        view = self.create_view(CatalogsView)
+        view.flash.assert_success_message('Catalog "{}" was saved'.format(name))
```
Comment (mshriver) [inline cfme/services/catalogs/catalog.py]: **Required**

A bit of scope creep, but I'd like to require this because its so small.

Please move the flash message validation to the test_catalog_crud test that's already being updated below, and only use `view.flash.assert_no_error` here.

This keeps a changing flash message from blocking/breaking tests that create a catalog item as a setup step.

File: cfme/tests/services/test_catalog.py [hunk 3]
```diff
@@ -6,46 +6,50 @@
...
-                  description="my catalog")
+    catalogs = []
+
+    def _create_catalog():
+        catalogs.append(appliance.collections.catalogs.create(name=fauxfactory.gen_alphanumeric(),
+                                                              description="my catalog"))
+
+    def _delete_catalog():
+        if catalogs:
+            catalogs[0].delete()
```
Comment (mshriver) [inline cfme/tests/services/test_catalog.py]: Why only delete the first one here, and not iterate over catalogs? (If there's a good reason put in-line in comment please :) )

File: cfme/rest/gen_data.py [hunk 4]
```diff
@@ -42,10 +41,11 @@ def service_catalogs(request, rest_api, num=5):
     return _creating_skeleton(request, rest_api, 'service_catalogs', scls_data, col_action='add')
 
 
-def service_catalog_obj(request, rest_api):
+def service_catalog_obj(appliance, request, rest_api):
```
Comment (psav) [inline cfme/rest/gen_data.py]: **REQUIRED:** As we are changing this anyway, can we drop the rest_api argument. when we invoke ```service_catalogs```, we can just pass, ```appliance.rest_api``` instead.

**Comment by psav** [comment]
Sorry @rsnyman just seen that the function you have changed the prototype of is also used in two other files:
```service_catalog_obj``` is used in ```cfme/services/fixtures.py``` and ```cfme/test/services/test_rest_services```
